dbapi/src/main/java/com/griddynamics/jagger/dbapi/provider/SessionInfoProviderImpl.java

```diff
@@ -273,53 +283,61 @@ public Long getFirstPosition(Set<String> selectedIds) throws RuntimeException {
             if (sessionDataList.isEmpty()) {
                 return Collections.<SessionDataDto>emptyList();
             }
 
             Map<Long, String> userCommentMap = Collections.EMPTY_MAP;
-
             if (isUserCommentStorageAvailable) {
-
-                List<Object[]> userComments = entityManager.createQuery(
-                        "select smd.sessionData.id, smd.userComment from SessionMetaDataEntity as smd where smd.sessionData in (:sessionDataList)")
-                        .setParameter("sessionDataList", sessionDataList)
-                        .getResultList();
-
+                query = entityManager.createQuery(
+                        "select smd.sessionData.id, smd.userComment from SessionMetaDataEntity as smd "
+                        + (isFetchAll ? " " : "where smd.sessionData in (:sessionDataList)")
+                );
+                if (!isFetchAll) {
+                    query.setParameter("sessionDataList", sessionDataList);
+                }
+                List<Object[]> userComments = query.getResultList();
                 if (!userComments.isEmpty()) {
                     userCommentMap = new HashMap<Long, String>(userComments.size());
                     for (Object[] objects : userComments) {
                         userCommentMap.put((Long) objects[0], (String) objects[1]);
                     }
                 }
             }
-            Multimap<Long, TagDto> tagMap = HashMultimap.create();
-
-            Set<Long> ids = new HashSet<Long>();
-            for (SessionData sd : sessionDataList) {
-                ids.add(sd.getId());
-            }
-
 
+            Multimap<Long, TagDto> tagMap = HashMultimap.create();
             if (isTagsStorageAvailable) {
-                List<Object[]> sessionTags = entityManager.createNativeQuery("select a.sessions_id, a.tags_name, te.description " +
-                        "from  TagEntity as te, (select distinct ste.sessions_id, ste.tags_name from SessionTagEntity as ste where ste.sessions_id in (:ids)) as a " +
-                        "where a.tags_name=te.name")
-                        .setParameter("ids", ids)
-                        .getResultList();
+                query = entityManager.createNativeQuery(
+                        "select a.sessions_id, a.tags_name, te.description " +
+                        "from  TagEntity as te, (select distinct ste.sessions_id, ste.tags_name "
+                        + "from SessionTagEntity as ste"
+                        + (isFetchAll ? " " : " where ste.sessions_id in (:ids)) as a ")
+                        + "where a.tags_name=te.name"
+                );
+                if (!isFetchAll) {
+                    Set<Long> ids = new HashSet<Long>();
+                    for (SessionData sd : sessionDataList) {
+                        ids.add(sd.getId());
+                    }
+                    query.setParameter("ids", ids);
+                }
```

Comment:
The methods getByDatePeriod, getBySessionIds and getBySessionTagsName seems to be too long for me. I'd recommend to split them in smaller private methods to make code cleaner and more readable (Rule of 30)